src/resultsdisassemblypage.cpp

```diff
@@ -123,6 +136,39 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(QWidget* parent)
             }
         }
     });
+
+#if KF5SyntaxHighlighting_FOUND
+    QStringList schemes;
+
+    for (const auto& entry : m_repository->definitions()) {
+        schemes.push_back(entry.name());
+    }
+
+    auto connectCompletion = [schemes, this](QComboBox* box, auto* model) {
+        auto m = new QStringListModel(this);
+        m->setStringList(schemes);
+
+        auto completer = new QCompleter(this);
+        completer->setModel(m);
+        completer->setCaseSensitivity(Qt::CaseInsensitive);
+        completer->setCompletionMode(QCompleter::PopupCompletion);
+        box->setCompleter(completer);
+
+        connect(completer, qOverload<const QModelIndex&>(&QCompleter::activated), this,
+                [this, model](const QModelIndex& entry) {
+                    model->setSyntaxHighlightDefinition(m_repository->definitionForName(entry.data().toString()));
+                });
+        connect(box, qOverload<int>(&QComboBox::activated), this, [this, model, box]() {
+            model->setSyntaxHighlightDefinition(m_repository->definitionForName(box->currentText()));
```

Comment:
when a file is opened, it will be set to file extension + ASM
the code above only triggers if the user sets the syntax highlighting to something custom